payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
@sbrossie yes, does make sense.

Adding a new boolean parameter and running all tests I found this particular case: [Can add a chargeback for deleted payment methods](https://github.com/killbill/killbill/blob/master/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestChargeback.java#L89-L89). Is that correct? Can a payment transaction be created over a deleted payment method?